### PR TITLE
feat(core): Store remote models on file system

### PR DESCRIPTION
**Pull request by mttrbrts**
# Issue #66
Support for offline download of external models

### Changes
- Adds `getModels` function to `ModelManager`
- Adds `writeModelsToFileSystem ` function to `ModelManager`
- Exposes `saveExternal` function in `concerto-cli`

### Flags
- Also fixes a bug with the `api.txt` generation when `hasInstance` was listed as `undefined`

### Related Issues
- Issue #66

### CODE REVIEW

File: packages/concerto-cli/index.js [hunk 1]
```diff
@@ -54,6 +54,28 @@ require('yargs')
                 console.log(err.message + ' ' + err);
             });
     })
+    .command('saveExternal', 'save local copies of external model dependencies', (yargs) => {
```
Comment (jeromesimeon) [inline packages/concerto-cli/index.js]: @dselman and I had suggested `concerto get` which I think is what the `go` compiler uses. https://golang.org/cmd/go/

Other options might be `concerto install` `concerto download` ...

File: packages/concerto-cli/index.js [hunk 2]
```diff
@@ -54,6 +54,28 @@ require('yargs')
...
+        yargs.option('ctoFiles', {
+            describe: 'array of local CTO files',
+            type: 'string',
+            array: true,
+            default: '.'
+        });
+        yargs.option('outputDirectory', {
```
Comment (jeromesimeon) [inline packages/concerto-cli/index.js]: Some of our other tools use `--out` (not always for directories though) so maybe we could try and align?

File: packages/concerto-core/lib/codegen/javascriptparser.js [hunk 3]
```diff
@@ -251,12 +251,17 @@ class JavaScriptParser {
...
+                            if(!thing.key.name && thing.key.property.name){
+                                name = thing.key.property.name;
+                            }
+
                             commentData = commentData || [];
                             if (visibility === '+' || visibility === '~' || includePrivates) {
                                 const method = {
                                     visibility: visibility,
                                     returnType: returnType,
-                                    name: thing.key.name,
```
Comment (jeromesimeon) [inline packages/concerto-core/lib/codegen/javascriptparser.js]: I'm unclear about those changes, do they have to do with the `api.txt` generation?

Comment (mttrbrts) [reply packages/concerto-core/lib/codegen/javascriptparser.js]: Yes, that's right, this library is used to extract the signature of the API for generating `api.txt` automatically

**Pull request merged**